internal/agent/device/device.go

```diff
@@ -317,10 +330,13 @@ func (a *Agent) beforeUpdate(ctx context.Context, current, desired *v1alpha1.Dev
 		}
 	}
 
+	a.prefetchManager.RegisterOCICollector(a.appManager.CollectOCITargets(ctx))
 	if a.specManager.IsOSUpdate() {
-		if err := a.osManager.BeforeUpdate(ctx, current.Spec, desired.Spec); err != nil {
-			return fmt.Errorf("os: %w", err)
-		}
+		a.prefetchManager.RegisterOCICollector(a.osManager.CollectOCITargets(ctx))
```

Comment:
Won't this register the same collectors multiple times? I could have missed it, but I don't see where collectors are being removed in the prefetchManager.